client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -420,7 +422,23 @@ private WinRm getService() {
         if (winrm != null) {
             return winrm;
         } else {
-            return createService();
+
+            if (context != null) {
+                cleanupContext = false;
+            } else {
+                context = WinRmClientContext.newInstance();
```

Comment:
The synchronization here is a little scary looking - for example if there are several concurrent calls to `command(...)` on its first use, then we'll do multiple calls to `WinRmClientContext.newInstance()`. However, `createService` is synchronized so we'll only create one `WinRm winrm` object.

However, your changes LGTM. We can think about and perhaps improve the synchronization in a separate PR at some point.

p.s. I hate people incorrectly using double-checked locking in Java (which is what the code was doing before, and is still doing for initialising the `winrm` field)!